src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { faCaretDown, faTools } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { pages } from '../../assets/pages';
+
+import './NavBar.css';
+import NavBarTools from './NavBarTools';
+import NavBarLogo from './NavBarLogo';
+
+interface PlatformType {
+  id: string;
+  header: string;
+  style: string;
+}
+interface NavBarPropsType {
+  toggleMainWidth: () => void;
+  mini: Boolean;
+  platform: PlatformType;
+}
+
+const NavBar = ({ toggleMainWidth, mini, platform }: NavBarPropsType) => {
+  const [sideBarWidthClass, setSideBarWidthClass] = useState('sidebar-min');
+
+  const toggleSideBarWidth = (isMouseEvent: Boolean) => {
+    if (isMouseEvent) {
+      toggleMainWidth();
+      setSideBarWidthClass(mini ? 'sidebar-max' : 'sidebar-min');
+    }
+  };
+
+  return (
+    <div>
+      <nav
+        className={`sidebar ${platform.style} ${sideBarWidthClass}`}
+        role="navigation"
+        onMouseOver={() => toggleSideBarWidth(true)}
+        onFocus={() => toggleSideBarWidth(false)}
+        onMouseOut={() => toggleSideBarWidth(true)}
+        onBlur={() => toggleSideBarWidth(false)}
+      >
+        <NavBarLogo header={platform.header} />
+        <div className="navbar-items">
+          {pages
+            .filter(
+              (page) =>
+                platform.id === 'home' || page.tags.includes(platform.id)
```

Comment:
also can be in redux , some sort of relevantPlatforms  variable  instated of this unnecessary filtering